devices/HumanControlBoard/HumanControlBoard.cpp

```diff
@@ -132,36 +134,60 @@ bool HumanControlBoard::open(yarp::os::Searchable& config)
 
 bool HumanControlBoard::close()
 {
-    pImpl->dynamicsPort.close();
     return true;
 }
 
 void HumanControlBoard::run()
 {
-    // Get data from IHumanState interface
-    pImpl->nJoints = pImpl->iHumanState->getNumberOfJoints();
-    pImpl->jointNameList = pImpl->iHumanState->getJointNames();
+    // Get number of joints and joint names
+    if (pImpl->isHumanStateAttached)
+    {
+        pImpl->nJoints = pImpl->iHumanState->getNumberOfJoints();
+        pImpl->jointNameList = pImpl->iHumanState->getJointNames();
+    }
+    else if (pImpl->isHumanDynamicsAttached)
+    {
+        pImpl->nJoints = pImpl->iHumanDynamics->getNumberOfJoints();
+        pImpl->jointNameList = pImpl->iHumanDynamics->getJointNames();
+    }
+    else
+    {
+        yError() << LogPrefix << "No interface attached to the device.";
+        askToStop();
+        return;
+    }
 
-    std::vector<double> jointPositionsInterface = pImpl->iHumanState->getJointPositions();
-    std::vector<double> jointVelocitiesInterface = pImpl->iHumanState->getJointVelocities();
+    // Initialize buffers for measurements
+    std::vector<double> jointPositionsInterface(pImpl->nJoints, 0.0);
```

Comment:
Minor: The variable ending of `Interface` is a bit misleading. May be we can drop it as : `jointPositionsInterface` -------> `jointPositions`

Also for other variables

Reply:
or a suffix as `Data`

Reply:
Yes, I was also not very convinced by that variable naming. I have changed to `jointXXXData`